packages/api/models/granules.js

```diff
@@ -64,29 +64,48 @@ class Granule extends Manager {
    * @returns {Promise} an object showing the start of the re-ingest
    */
   async reingest(g) {
+    await this.applyWorkflow(g, 'IngestGranule', 'input');
+    return {
+      granuleId: g.granuleId,
+      action: 'reingest',
+      status: 'SUCCESS'
+    };
+  }
+
+  /**
+   * apply a workflow to a given granule object
+   *
+   * @param {Object} g - the granule object
+   * @param {string} workflow - the workflow name
+   * @param {string} messageSource - 'input' or 'output' from previous execution
+   * @param {Object} metaOverride - overrides the meta of the new execution
+   * @param {Object} payloadOverride - overrides the payload of the new execution
+   * @returns {Promise} an object showing the start of the workflow execution
```

Comment:
Usually when I know the final type of the promise, I add it such as `Promise<Object>`